src/frontend/packages/auth-provider/src/authProvider.js

```diff
@@ -1,36 +1,81 @@
-  if (![AUTH_TYPE_SSO, AUTH_TYPE_LOCAL, AUTH_TYPE_POD].includes(authType))
+const authProvider = ({
+  dataProvider,
+  authType,
+  allowAnonymous = true,
+  checkUser,
+  checkPermissions = false,
+  clientId
+}) => {
+  if (![AUTH_TYPE_SSO, AUTH_TYPE_LOCAL, AUTH_TYPE_POD, AUTH_TYPE_SOLID_OIDC].includes(authType))
     throw new Error('The authType parameter is missing from the auth provider');
+  if (authType === AUTH_TYPE_SOLID_OIDC && !clientId)
+    throw new Error('The clientId parameter is required for solid-oidc authentication');
   return {
     login: async params => {
-      const authServerUrl = await getAuthServerUrl(dataProvider);
-      if (authType === AUTH_TYPE_LOCAL) {
-        const { username, password } = params;
+      if (authType === AUTH_TYPE_SOLID_OIDC) {
+        const { webId, issuer } = params;
+
+        if (webId && !issuer) {
+          // TODO find issuer from webId
+        }
+
+        const as = await oauth
+          .discoveryRequest(new URL(issuer))
+          .then(response => oauth.processDiscoveryResponse(new URL(issuer), response));
+
+        const codeVerifier = oauth.generateRandomCodeVerifier();
+        const codeChallenge = await oauth.calculatePKCECodeChallenge(codeVerifier);
+        const codeChallengeMethod = 'S256';
+
+        // Save to use on handleCallback method
+        localStorage.setItem('code_verifier', codeVerifier);
+
+        const authorizationUrl = new URL(as.authorization_endpoint);
+        authorizationUrl.searchParams.set('response_type', 'code');
+        authorizationUrl.searchParams.set('client_id', clientId);
+        authorizationUrl.searchParams.set('code_challenge', codeChallenge);
+        authorizationUrl.searchParams.set('code_challenge_method', codeChallengeMethod);
+        authorizationUrl.searchParams.set('redirect_uri', `${window.location.origin}/auth-callback`);
+        authorizationUrl.searchParams.set('scope', 'openid webid offline_access');
+
+        window.location = authorizationUrl;
+      } else if (authType === AUTH_TYPE_LOCAL) {
+        const { username, password, interactionId, redirectTo } = params;
+        const authServerUrl = await getAuthServerUrl(dataProvider);
         try {
           const { json } = await dataProvider.fetch(urlJoin(authServerUrl, 'auth/login'), {
             method: 'POST',
             body: JSON.stringify({
               username: username.trim(),
-              password: password.trim()
+              password: password.trim(),
+              interactionId
             }),
             headers: new Headers({ 'Content-Type': 'application/json' })
           });
           const { token } = json;
           localStorage.setItem('token', token);
-          // Reload to ensure the dataServer config is reset
-          window.location.reload();
+          if (redirectTo) {
+            if (interactionId) await delay(3000); // Ensure the interactionId has been received and processed
```

Comment:
Should this be `if (!interactionId)...`?

Reply:
No, we add a delay if there is an interaction with the OIDC provider, because after login the OIDC provider must update its data.